Source/CarthageKit/Project.swift

```diff
@@ -187,11 +187,19 @@ public final class Project {
 		}
 
 		return cartfile.zipWith(privateCartfile)
-			.map { (var cartfile, privateCartfile) -> Cartfile in
-				cartfile.appendCartfile(privateCartfile)
+			.map { (var cartfile, privateCartfile) -> ColdSignal<Cartfile> in
+				// TODO: use Set when Carthage supports Swift 1.2
+				let duplicateDeps = filter(privateCartfile.dependencies) { privateDep in
+					contains(cartfile.dependencies, {$0.project == privateDep.project})
+				}
```

Comment:
I’ll write two functions: one that lists dupes in an array and another that lists dupes in two different arrays. What’s the best place to put them in the code base: as generic code in a new file with extensions to the standard library, or as methods in struct Cartfile?

I’ve thought of using Foundation for this, but that would mean converting struct ProjectIdentifier to an objc class.